src/plugins_types/enumeration.c

```diff
@@ -71,6 +71,13 @@ lyplg_type_store_enum(const struct ly_ctx *ctx, const struct lysc_type *type, co
                     *(int32_t *)value);
             goto cleanup;
         }
+
+        if (options & LYPLG_TYPE_STORE_DYNAMIC) {
+            free((void *)value);
+            options &= ~LYPLG_TYPE_STORE_DYNAMIC;
+        }
```

Comment:
What is the purpose of this code here? It is always executed in cleanup.

Reply:
yes, it could be done better. It will be fixed.